doc/tutorials/cascading.md

```diff
@@ -0,0 +1,131 @@
+
+This tutorial assumes that you know what namespaces are. You might want to read
+the [Namespaces tutorial](/doc/tutorials/namespaces.md) first.
+
+In Elektra, the default order of namespaces is as follows:
+
+ * [spec](https://github.com/ElektraInitiative/libelektra/blob/master/doc/help/elektra-namespaces.md#spec) (modify elektra lookup behaviour)
+ * [proc](https://github.com/ElektraInitiative/libelektra/blob/master/doc/help/elektra-namespaces.md#proc) (process-related information)
+ * [dir](https://github.com/ElektraInitiative/libelektra/blob/master/doc/help/elektra-namespaces.md#dir) (directory-related information, e.g. `.git` or `.htaccess`)
+ * [user](https://github.com/ElektraInitiative/libelektra/blob/master/doc/help/elektra-namespaces.md#user) (user configuration)
+ * [system](https://github.com/ElektraInitiative/libelektra/blob/master/doc/help/elektra-namespaces.md#system) (system configuration)
+ * [cascading](https://github.com/ElektraInitiative/libelektra/blob/master/doc/help/elektra-namespaces.md#cascading) (keys without a namespace)
+
+Looking at this order, we can see that if a configuration option isn't specified
+by the user (then it would be in the `user` namespace), it will be loaded from
+the `system` namespace. In this case, the option in the `system` namespace works
+like a default value.
+
+```
+$ sudo kdb set system/t/test "hello world"
+create a new key system/t/test with string hello world
+
+$ kdb get /t/test
+hello world
+
+$ kdb set user/t/test "hello universe"
+Create a new key user/t/test with string hello universe
+
+$ kdb get /t/test
+hello universe
+```
+
+Furthermore, we can see that `dir` is even higher than `user`, which means that
+configuration in the current folder can overwrite user configuration.
+
+`.configuration` in your current directory:
+
+```
+test = hello dir
+```
+
+Then run:
+
+```
+$ sudo kdb mount /.configuration dir/t ini
+$ kdb get /t/test
+hello dir
+```
+
+
+## Cascading
+
+Cascading triggers secondary actions when, for example, the key isn't found.
+This concept is used for our previous example of using `system` configuration as
+default values. When a key starts with `/`, *cascading lookup* will
+automatically be performed. e.g. using `/test` instead of `system/test` will do
+a cascading lookup.
+
+
+## Override links
+
+The `spec` namespace is special as it can completely change how the cascading
+lookup works.
+
+For example, in the meta data of the respective `spec`-keys, *override links*
+can be specified to use other keys in favor of the key itself. This way, even
+config from current folders (`dir`) can be overwritten.
+
+In the cascading lookup, override links come in as follows:
+
+ 1. `override/#` keys will be considered
+ 2. namespaces specified in the `namespaces/#` keys are considered
+ 3. Otherwise, all namespaces will be considered, see [here](/doc/help/elektra-namespaces.md).
+ 4. `fallback/#` keys will be considered
+ 5. `default` value will be returned
```

Comment:
I never mentioned that they are, and changed the text before the list to be more specific. I also explained what the `#` means below.